src/frontend/src/lib/stores/modal.store.ts

```diff
@@ -68,38 +69,47 @@ export interface ModalStore<T> extends Readable<ModalData<T>> {
 	close: () => void;
 }
 
-const initModalStore = <T>(): ModalStore<T> => {
+export const initModalStore = <T>(): ModalStore<T> => {
 	const { subscribe, set } = writable<ModalData<T>>(undefined);
 
+	const setType = (type: Modal<T>['type']) => () => set({ type });
```

Comment:
Does this work? If I get it right, you are declaring function that returns function but, often we consume the store directly in the component so I'm wondering if the function are actually triggered?

For example

```
<button on:click={modalStore.openSend} />
```

Reply:
Apparently it works: since the return is a function of the same type, so it runs it as function.

BTW, this gave me the idea, that we can create a simple E2E test for all the Hero buttons, without necessary following the flow

![Screenshot 2024-11-18 at 08 44 08](https://github.com/user-attachments/assets/04bfd00d-44bc-43ac-8d68-60b1737e837e)

https://github.com/user-attachments/assets/34e28e9d-2d90-4a35-82b7-dad6ceb90f42

Reply:
> Apparently it works: since the return is a function of the same type, so it runs it as function.

Ah cool! Thanks for the explanation.

> BTW, this gave me the idea, that we can create a simple E2E test for all the Hero buttons, without necessary following the flow

That would be cool!